101-batch-with-keyvault-pfx-password/prereqs/prereq.azuredeploy.json

```diff
@@ -0,0 +1,74 @@
+﻿{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "certificatePrefixName": {
+      "type": "string"
+    },
+    "certificatePfxBase64": {
+      "type": "securestring"
+    },
+    "certificatePfxPassword": {
+      "type": "string"
+    }
+  },
+  "variables": {
+    "storagename": "[concat('storage', uniqueString(resourceGroup().id))]",
+    "batchname": "[concat('batch', uniqueString(resourceGroup().id))]",
+    "locationname": "[resourceGroup().location]",
+    "keyvaultname": "[concat('keyvault', uniqueString(resourceGroup().id))]",
+    "tenantId": "[subscription().tenantId]",
+    "certificateName": "[concat(parameters('certificatePrefixName'), uniqueString(resourceGroup().id))]"
+  },
+  "resources": [
+    {
+      "apiVersion": "2015-08-01",
+      "name": "[variables('certificateName')]",
+      "type": "Microsoft.Web/certificates",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "pfxBlob": "[parameters('certificatePfxBase64')]",
+        "password": "[parameters('certificatePfxPassword')]"
+      },
+      "tags": {
+        "displayName": "Certificate"
+      }
+    },
+    {
+      "apiVersion": "2016-01-01",
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('storagename')]",
+      "location": "[variables('locationname')]",
+      "sku": {
+        "name": "Standard_LRS"
+      },
+      "kind": "Storage"
+    },
+    {
+      "type": "Microsoft.Batch/batchAccounts",
+      "apiVersion": "2015-12-01",
+      "name": "[variables('batchname')]",
+      "location": "[variables('locationname')]",
+      "properties": {
+        "autoStorage": {
+          "storageAccountId": "[resourceId('Microsoft.Storage/storageAccounts', variables('storagename'))]"
+        }
+
+      },
+      "dependsOn": [
+        "[resourceId('Microsoft.Storage/storageAccounts', variables('storagename'))]"
+      ]
+
+    }    
+  ],
+  "outputs": {
+    "batchservice": {
+      "type": "string",
+      "value": "[variables('batchname')]"
+    },
+    "certthumbprint": {
+      "type": "string",
+      "value": "[reference(resourceId('Microsoft.Web/certificates', variables('certificateName')), providers('Microsoft.Web', 'certificates').apiVersions[0]).thumbprint]"
```

Comment:
remove providers() call and use a literal apiVersion

Reply:
Done. Removed provider call and used literal apiVersion.
      "value": "[reference(resourceId('Microsoft.Web/certificates', variables('certificateName')), '2018-02-01').thumbprint]"